shapely/geos.py

```diff
@@ -257,6 +257,8 @@ def __del__(self):
 
     def read(self, text):
         """Returns geometry from WKT"""
+        if not isinstance(text, (bytes, str) if sys.version_info[0] >= 3 else (str, unicode)):
+            raise TypeError("Only bytes or string types can be processed.")
```

Comment:
The problem here isn't Shapely's WKT reader, it's ctypes' c_char_p, which interprets integers as pointers to memory. A `if isinstance(text, int): raise TypeError(...)` will suffice – we only need to intercept the int case. See https://bugs.python.org/issue1701409 for more details.

Reply:
in this case there are another issue for python3 https://github.com/Toblerity/Shapely/blob/master/shapely/geos.py#L261 which will raise `AttributeError`

Reply:
I agree 👍

Reply:
so what the best approach for this bug fixing:
1. use byte/string instance parameter check?
2. use int parameter check + add AttributeError handling and TypeError reraising?

Reply:
I'm sorry that I wasn't clear: number 2, please.

Reply:
one more question, for python 3 `bytes` type doesn't allowed now, should be this case covered or only unicode string should be allowed?

Reply:
added bytes handling for python3

Reply:
updated